### PR TITLE
Emergency shutdown implementation

### LINKED ISSUE #91: Set an emergency shutdown address instead of using owner address
Add a `terminator` address instead of Controller's owner address, that can trigger the emergency shutdown.

### PR DESCRIPTION
**Pull request by haythem96**
# Emergency shutdown implementation

## High Level Description

This PR implement the emergency shutdown mechanism.
Closes #91 .

### Code

- [x] Unit test 100% coverage
- [x] Does your code follow the naming and code documentation guidelines?

### Documentation

- [x] Is your code up to date with the spec? 
- [x] Have you added your tests to the testing doc?

### CODE REVIEW

**Review by antoncoding — APPROVED** [review]
Looks good.

File: test/controller.test.ts [hunk 1]
```diff
@@ -3250,20 +3250,300 @@ contract('Controller', ([owner, accountOwner1, accountOwner2, accountOperator1,
...
+
+    it('should settle vault when system is paused', async () => {
+      // past time after expiry
+      await time.increase(60 * 61) // increase time with one hour in seconds
+      // set price in Oracle Mock, 150$ at expiry, expire ITM
+      await oracle.setExpiryPrice(
+        await shortOtoken.underlyingAsset(),
+        new BigNumber(await shortOtoken.expiryTimestamp()),
+        new BigNumber(150).times(new BigNumber(10).exponentiatedBy(18)),
+      )
```
Comment (aparnakr) [inline test/controller.test.ts:3458]: shouldn't there be a separate function for this?

Comment (aparnakr) [reply test/controller.test.ts:3458]: Why comine this and redeemForEmergency?

Comment (aparnakr) [reply test/controller.test.ts:3458]: What happens when the otoken has not expired? How will a seller / buyer be able to get their share of $?

**Review by aparnakr — CHANGES_REQUESTED** [review]
Hmm, how will a seller / buyer be able to get money out of the system if we pause before the short token expires?

**Review by aparnakr — APPROVED** [review]
LGTM

**Pull request merged**